src/test/aws/construct/static-asset-deployment.test.ts

```diff
@@ -24,89 +24,135 @@ const testStackProps = {
+  app: App,
+  Construct: new (parent: Construct, id: string, props: P) => T,
+  id: string,
+  appProps: P
+): CommonStack {
+  class InitTestCommonStack extends CommonStack {
+    constructor(parent: cdk.App, id: string, props: P) {
+      super(parent, id, props)
 
-  protected determineConstructProps(props: cdk.StackProps) {
-    return {
-      ...super.determineConstructProps(props),
-      staticAssetBucket: this.node.tryGetContext('siteBucket'),
-      staticAssetDeployment: {
-        prune: true,
-        retainOnDelete: false,
-      },
-      staticAssetSources: [s3deploy.Source.asset('src/test/aws/common/resources/')],
-      staticAssetsForExport: [{ key: 'myCSV', value: 'test.csv' }],
+      this.construct = new Construct(this, id, { ...this.props, ...props })
+      // ideally this would be part of
+      if ('initResources' in this.construct && typeof this.construct.initResources === 'function') {
+        this.construct.initResources()
+      }
     }
   }
+
+  return new InitTestCommonStack(app, id, appProps)
 }
 
-class TestStaticAssetDeployment extends StaticAssetDeployment {
-  declare props: TestStackProps
+const createStack = (
+  context: typeof testContext,
+  props: StaticAssetDeploymentProps
+): { stack: CommonStack; template: Template } => {
+  const app = new cdk.App({ context })
+  const stack = createTestStack(app, StaticAssetDeployment, 'test-static-asset-deployment-stack', props)
 
-  constructor(parent: Construct, id: string, props: TestStackProps) {
-    super(parent, id, props)
-    this.props = props
-    this.id = 'test-static-asset-deployment'
-    this.initResources()
-  }
+  const template = Template.fromStack(stack)
+  return { stack, template }
 }
 
-const app = new cdk.App({ context: testStackProps })
-const stack = new TestCommonStack(app, 'test-static-asset-deployment-stack', testStackProps)
-const template = Template.fromStack(stack)
+// name, stack and expected
+type TestTuple = [string, { stack: CommonStack; template: Template }, { bucketCount: number }]
 
-describe('TestStaticAssetDeploymentConstruct', () => {
-  test('is initialised as expected', () => {
-    /* test if the created stack have the right properties injected */
-    expect(stack.props).toHaveProperty('staticAssetDeployment')
-    expect(stack.props.staticAssetDeployment.prune).toEqual(true)
-  })
-})
+const defaultProps = {
+  staticAssetBucket: {
+    bucketName: 'site',
+    autoDeleteObjects: true,
+    removalPolicy: 'destroy',
+    serverAccessLogsPrefix: 'logs/',
+    websiteIndexDocument: 'index.html',
+    websiteErrorDocument: 'index.html',
+    existingBucket: false,
+  },
+  staticAssetDeployment: {
+    prune: true,
+    retainOnDelete: false,
+  },
+  staticAssetSources: [s3deploy.Source.asset('src/test/aws/common/resources/')],
+  staticAssetsForExport: [{ key: 'myCSV', value: 'test.csv' }],
+} as any as StaticAssetDeploymentProps
+
+describe.each<TestTuple>([
```

Comment:
`table test` that allows us to test different configuration variations of a construct